test/fixtures/rules/std_rule_cases/L027.yml

```diff
@@ -73,3 +73,30 @@ test_ignore_value_table_functions:
   configs:
     core:
       dialect: bigquery
+
+test_fail_column_and_alias_same_name:
+  # See issue #2169
+  fail_str: |
+    SELECT
+        foo AS foo,
+        bar AS bar
+    FROM
+        a LEFT JOIN b ON a.id = b.id
+
+test_pass_column_and_alias_same_name_1:
+  pass_str: |
+    SELECT
+        a.foo AS foo,
+        b.bar AS bar
+    FROM
+        a LEFT JOIN b ON a.id = b.id
```

Comment:
Should we add similar test cases for t-sql and mysql since you had to explicitly add extra code for them?

Any ideas how we can ensure no one forgets to add a `get_alias` for any future dialect that implements it's own `SelectClauseElementSegment`?

Reply:
@tunetheweb I've added the extra unit tests.

For the `get_alias` in future dialects point, I think that's something we'd pick up in code review, since I'd expect the reviewer to look at the segment being overwritten? Don't think we have anything special in place for the existing similar `get_eventual_aliases`/`AliasInfo` cases.

Reply:
Yeah couldn’t think of any systematic way of checking but thought worth asking in case wiser minds could.